streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableRightJoin.java

```diff
@@ -84,20 +88,30 @@ public void process(final K key, final Change<V1> change) {
             }
 
             final R newValue;
+            final long resultTimestamp;
             R oldValue = null;
 
-            final V2 value2 = valueGetter.get(key);
-            if (value2 == null) {
+            final ValueAndTimestamp<V2> valueAndTimestampLeft = valueGetter.get(key);
+            final V2 valueLeft = getValueOrNull(valueAndTimestampLeft);
+            if (valueLeft == null) {
                 return;
             }
 
-            newValue = joiner.apply(change.newValue, value2);
+            if (change.newValue != null) {
+                resultTimestamp = Math.max(context().timestamp(), valueAndTimestampLeft.timestamp());
+            } else {
+                resultTimestamp = valueAndTimestampLeft.timestamp();
```

Comment:
Similar question here. Even if the left value is null, should the `null` record's timestamp still contribute to the join result time?

Reply:
Note, that this processor implement the right-hand-side input for a left-join. Hence, `change.newValue` is an update to the right hand side. If it's `null`, we get `<key:left-null>` as result and the current reasoning is that this should use the timestamp from the left hand side, similar to outer join. However, following you other comment, I agree that it might make sense to actually use the tombstone timestamp for the result in this case. Thought?

Reply:
SGTM. Thanks! (and thanks for the reasoning as well)

Reply:
Sure. As you might have noticed, I change this to use the tombstone timestamp (ie, `context.timestamp()`, now. Similar to https://github.com/apache/kafka/pull/6725#discussion_r284432739